gcsio/src/main/java/com/google/cloud/hadoop/gcsio/authorization/StorageRequestSummary.java

```diff
@@ -0,0 +1,81 @@
+package com.google.cloud.hadoop.gcsio.authorization;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Encapsulation of StorageRequest.
+ *
+ * The encapsulation only contains resource related information.
+ * Including: action, GCS bucket name and object path.
+ */
+public class StorageRequestSummary {
+  private final ActionType actionType;
+  private final List<GcsStorage> peers;
```

Comment:
rename as `resources` - add remember to change `toString` and corresponding getter.